Mathlib/NumberTheory/FactorisationProperties.lean

```diff
@@ -0,0 +1,235 @@
+  abundant
+* `Nat.prime_deficient`: all prime natural numbers are deficient
+* `Nat.exists_infinite_deficients`: there are infinitely many deficient numbers
+* `Nat.prime_pow_deficient`: any natural number power of a prime is deficient
+
+## Implementation Details
+* Zero is not included in any of the definitions and these definitions only apply to natural
+  numbers greater than zero.
+
+## References
+* Prielipp, Robert W. “PERFECT NUMBERS, ABUNDANT NUMBERS, AND DEFICIENT NUMBERS.”
+  The Mathematics Teacher, vol. 63, no. 8, 1970, pp. 692–96. JSTOR,
+  http://www.jstor.org/stable/27958492. Accessed 21 Feb. 2024.
+
+## Tags
+
+abundant, deficient, weird, pseudoperfect
+-/
+
+open Nat BigOperators Finset
+
+namespace Nat
+
+variable (n m k : ℕ)
+
+/-- `n : ℕ` is Abundant if and only if the sum of the proper divisors of `n` is greater than `n`-/
+def Abundant (n : ℕ) : Prop := ∑ i in properDivisors n, i > n
+
+/--  `n : ℕ` is Deficient if and only if the sum of the proper divisors of `n` is less than `n`-/
+def Deficient (n : ℕ) : Prop := ∑ i in properDivisors n, i < n
+
+/-- `n : ℕ` is Pseudoperfect if and only if there exists a subset of the proper divisors of n such
+  that the sum of that subset is equal to `n` and `n` is positive -/
+def Pseudoperfect (n : ℕ) : Prop :=
+  0 < n ∧ ∃ s : Finset ℕ, s ∈ powerset (properDivisors n) ∧ ∑ i in s, i = n
+
+/-- `n : ℕ` is a weird number if and only if it is both abundant and pseudoperfect -/
+def WeirdNumber (n : ℕ) : Prop := Abundant n ∧ ¬ Pseudoperfect n
+
+theorem not_pseudoperfect_iff_forall :
+    ¬ Pseudoperfect n ↔ n = 0 ∨ ∀ (s : Finset ℕ), s ∈ powerset (properDivisors n)
+      → ∑ i in s, i ≠ n := by
+    have hn : (¬ 0 < n) ↔ n = 0 := by simp only [not_lt, nonpos_iff_eq_zero]
+    dsimp only [Pseudoperfect]
+    rw [not_and_or, not_exists, hn]
+    constructor
+    · rintro (h1 | h2)
+      · tauto
+      · right
+        intro s
+        have hs : ¬(s ∈ powerset (properDivisors n) ∧ ∑ i in s, i = n) := h2 s
+        rw [not_and] at hs
+        exact hs
+    · rintro (h1 | h2)
+      · tauto
+      · right
+        simp_rw [not_and]
+        exact h2
+
+theorem one_deficient : Deficient 1 := le.refl
+
+theorem two_deficient : Deficient 2 := le.refl
+
+theorem three_deficient : Deficient 3 := by
+  dsimp only [Deficient]
+  norm_num
+
+theorem six_perfect : Perfect 6 := by
+  dsimp only [Perfect]
+  rw [show properDivisors 6 = {1, 2, 3} by rfl]
+  norm_num
+
+theorem twelve_abundant : Abundant 12 := by
+  dsimp only [Abundant]
+  rw [show properDivisors 12 = {1,2,3,4,6} by rfl]
+  norm_num
+
+set_option maxRecDepth 2000 in
+theorem seventy_weird : WeirdNumber 70 := by
+  dsimp only [WeirdNumber, Abundant]
+  rw [not_pseudoperfect_iff_forall]
+  have div70 : properDivisors 70 = {1, 2, 5, 7, 10, 14, 35} := by rfl
+  constructor
+  · rw [div70]
+    repeat norm_num
+  · rw [div70]
+    right
+    intro s hs
+    fin_cases hs <;> decide
+
+lemma deficient_not_abundant_or_perfect (hn : 0 < n) :
```

Comment:
Is this really needed?

Reply:
I wanted to demonstrate that a deficient number could never be abundant or perfect. I wanted to show that exclusivity for the other factorisation properties as well.

Reply:
The thing we'd like to have here is "for a natural number n, exactly one of the following is true: n=0, n is deficient, n is abundant, n is perfect". Like David I'm not convinced that these lemmas are the right way to say this, but on the other hand it would be nice to be able to express this idea in a useful way.

Reply:
Would an xor work?

Reply:
I think that what we have right now will do.